src/plugins/auto_batch/src/async_infer_request.cpp

```diff
@@ -54,6 +54,14 @@ std::vector<ov::ProfilingInfo> AsyncInferRequest::get_profiling_info() const {
         return m_request_without_batch->get_profiling_info();
 }
 
+std::vector<ov::SoPtr<ov::IVariableState>> AsyncInferRequest::query_state() const {
+    check_state();
+    if (SyncInferRequest::eExecutionFlavor::BATCH_EXECUTED == m_sync_request->m_batched_request_status)
```

Comment:
Is there thread race condition risk? because m_sync_request->m_batched_request_status maybe updated in a separated thread.

Reply:
I think no. Before query, it will be locked at:
https://github.com/openvinotoolkit/openvino/blob/b2d361fcca75d419b15e55b8db3fb34eb3fc7e4c/src/inference/src/dev/iasync_infer_request.cpp#L192-L202

Reply:
This mutex will be unlocked after return from check_state() method

Reply:
shall be no race issue here from my point of view
only that the profiling info /state status will not be the data for the batched infer request instead of individual request.
better to understand the usage scenario to decide if we need to match the batched profilinginfo /status back to individual profiling info/ state info.